homework/index.js

```diff
@@ -30,15 +30,106 @@
+        const navTitle = createAndAppend('div', navHeader, { class: 'nav-title' });
+        navTitle.innerText = 'HYF Repositories';
+        const selectEl = createAndAppend('select', navDiv, { id: 'getRepoData' });
+        data.sort((a, b) => (a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1));
+        data.forEach(res => {
+          createAndAppend('option', selectEl, { text: res.name, value: res.id });
+        });
+      })
+      .catch(err => {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+      });
+  }
+
+  function getRepoData(url) {
+    const selectRepo = document.getElementById('getRepoData');
+    const container = createAndAppend('div', root, { class: 'container' });
+    function fetchSingleData() {
+      return fetch(url)
+        .then(res => res.json())
+        .then(repData => {
+          container.innerHTML = '';
+          const filteredData = repData.filter(el => el.id === Number(selectRepo.value));
+          const repositoriesSection = createAndAppend('div', container, { class: 'repInfo-side' });
+          const repositoryName = createAndAppend('div', repositoriesSection, { class: 'repName' });
+          const repositoryDescription = createAndAppend('div', repositoriesSection, {
+            class: 'repDescription',
+          });
+          const repositoryForks = createAndAppend('div', repositoriesSection, {
+            class: 'repForks',
+          });
+          const repositoryDate = createAndAppend('div', repositoriesSection, { class: 'repDate' });
+          filteredData.forEach(allData => {
+            createAndAppend('span', repositoryName, { text: 'Repository Name:' });
+            createAndAppend('a', repositoryName, {
+              text: allData.name,
+              href: allData.html_url,
+              target: '_blank',
+            });
+            createAndAppend('span', repositoryDescription, { text: 'Description:' });
+            createAndAppend('p', repositoryDescription, { text: allData.description });
+            createAndAppend('span', repositoryForks, { text: 'Forks:' });
+            createAndAppend('p', repositoryForks, { text: allData.forks });
+            createAndAppend('span', repositoryDate, { text: 'Date:' });
+            createAndAppend('p', repositoryDate, { text: allData.updated_at });
+
+            const contributorSection = createAndAppend('div', container, {
+              class: 'ContInfo-side',
+            });
+            createAndAppend('h4', contributorSection, {
+              text: 'Contributions',
+              class: 'contributorHeader',
+            });
+            const contributorWrapper = createAndAppend('div', contributorSection, {
+              class: 'contributorWrapper',
+            });
+            return fetch(allData.contributors_url)
```

Comment:
I think you should try to move fetching contributors data to its own separate function and then call that function from here and from the event listener. Please have a look into my comment on the Slack class19 channel and try to refactor this code.